### DISCUSSION TITLE
Domain name as dynamic DNS for passive mode

### DISCUSSION THREAD

**Discussion by RainOne** [drakkan/sftpgo#1430] [General]
Hey, I've been researching this scenario for a while and I can't find any direct solutions.

I have a sFTPGo instance running inside an Unraid machine, through their docker application, behind a NAT. 
I, obviously, can only get external passive mode to work if I input my external ip address on the SFTPGO_FTPD__BINDINGS__0__FORCE_PASSIVE_IP setting on the docker - so far so good.

If I don't input anything, it detects the internal Docker IP address (which wouldn't work for my local network). If I input my local ip address on that machine, Passive works internally, but obviously not externally. If I input my external IP, it works externally - everything working as intended so far.

I read a support ticket that said that Dynamic Domain support would be added on 4.5.0, and it seems like that indeed it was. However, if I put a domain name on the SFTPGO_FTPD__BINDINGS__0__FORCE_PASSIVE_IP variable, the  service just won't start - the log just tells me that the application errors out:

{"level":"error","time":"2023-10-01T18:12:16.981","sender":"service","message":"could not start FTP server: could not load settings: the provided passive IP \"{DOMAIN_GOIES_HERE}\" is not valid"}

The message is provided exactly as is (I obivously changed the domain name with {DOMAIN_GOIES_HERE}, but the slashes and " are there on the message.

Do I need a different setting to pass a domain name as the external IP? I can't seem to figure this one out.

**Comment by stanthewizzard** [comment]
Same issue and interested by a solution

**Comment by IAmChills** [comment]
This is still an issue for me.

**Comment by ZP1314ace** [comment]
I met the same problem

**Comment by drakkan** [comment]
Please try setting the passive host; it accepts a domain name

**Reply by tkrn** [reply]
This worked for me on TrueNAS Scale Apps. I was able to set the passive env variable to an ip address but the trick for me was to use "host networking" on the TrueNAS side for the container.
<img width="778" height="819" alt="Screenshot from 2026-04-23 13-14-14" src="https://github.com/user-attachments/assets/44791e85-da21-4b76-8986-a8c920f5f41f" />
